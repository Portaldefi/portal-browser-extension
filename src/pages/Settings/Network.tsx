import React, { useCallback } from 'react';
import {
  // Button, 
  Grid, Header,
  // Icon, 
  List
} from 'semantic-ui-react';
import { useNavigate } from 'react-router-dom';
import ConnectionItem from '../../components/Home/TransactionHistory/TransactionItem';

import NetworkOption from '../../components/Settings/NetworkOption';

export default () => {
  const navigate = useNavigate();

  const handleClickConnection = useCallback(() => {
    navigate('/connection-detail')
  }, []);

  return (
    <Grid className='w-100'>
      <Grid.Row centered className='p-none pb-19'>
        <Header as='h1' className='heading'>Settings</Header>
      </Grid.Row>

      <Grid.Row className='p-none'>
        <List className='w-100'>
          <List.Item onClick={handleClickConnection}>
            <List.Content>
              <ConnectionItem name='Bitcoin' comment='https://grove.chat/  -  Tue, Apr. 26, 2022 5:00 am' />
            </List.Content>
          </List.Item>
          <List.Item onClick={handleClickConnection}>
            <List.Content>
              <ConnectionItem name='Ethereum' comment='https://uniswap.org/  -  Mon, Apr. 25, 2022 8:00 pm' />
            </List.Content>
          </List.Item>
        </List>
      </Grid.Row>
    </Grid>
  );
}