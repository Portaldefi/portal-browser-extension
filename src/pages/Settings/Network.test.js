import Network from './Network';
import { create, act } from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store'

describe('NetworkComponent', () => {
  const initialState = {
    menu: {
      items: []
    },
    key: {
      privateKey: '',
      privateExtendedKey: '',
      identity: [],
      selectedIdentityId: 0,
      selectedChainId: 0
    }
  }
  const mockStore = configureStore()
  let store


  it('should work', async () => {
    let tree;
    act(() => {
      store = mockStore(initialState)
      tree = create(
        <Provider store={store}>
          <BrowserRouter>
            <Network />
          </BrowserRouter>
        </Provider>
      );
    });

    expect(tree).toMatchSnapshot();
  });
  afterAll(() => jest.resetModules());
});