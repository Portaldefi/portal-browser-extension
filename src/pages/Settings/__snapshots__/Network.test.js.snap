// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkComponent should work 1`] = `
<div
  className="ui grid w-100"
>
  <div
    className="centered row p-none pb-19"
  >
    <h1
      className="ui header heading"
    >
      Settings
    </h1>
  </div>
  <div
    className="row p-none"
  >
    <div
      className="ui list w-100"
      role="list"
    >
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="content"
        >
          <div
            className="ui grid p-16 py-9 bg-white"
          >
            <div
              className="left aligned eleven wide column"
            >
              <p
                className="ui header description"
              >
                Bitcoin
              </p>
              <p
                className="ui header comment"
              >
                https://grove.chat/  -  Tue, Apr. 26, 2022 5:00 am
              </p>
            </div>
            <div
              className="middle aligned five wide column"
            >
              <div
                className="ui fitted toggle checkbox"
                onChange={[Function]}
                onClick={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
              >
                <input
                  checked={false}
                  className="hidden"
                  readOnly={true}
                  tabIndex={0}
                  type="checkbox"
                />
                <label />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="content"
        >
          <div
            className="ui grid p-16 py-9 bg-white"
          >
            <div
              className="left aligned eleven wide column"
            >
              <p
                className="ui header description"
              >
                Ethereum
              </p>
              <p
                className="ui header comment"
              >
                https://uniswap.org/  -  Mon, Apr. 25, 2022 8:00 pm
              </p>
            </div>
            <div
              className="middle aligned five wide column"
            >
              <div
                className="ui fitted toggle checkbox"
                onChange={[Function]}
                onClick={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
              >
                <input
                  checked={false}
                  className="hidden"
                  readOnly={true}
                  tabIndex={0}
                  type="checkbox"
                />
                <label />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`NetworkComponent should work 2`] = `
<div
  className="ui grid w-100"
>
  <div
    className="centered row p-none pb-19"
  >
    <h1
      className="ui header heading"
    >
      Settings
    </h1>
  </div>
  <div
    className="row p-none"
  >
    <div
      className="ui list w-100"
      role="list"
    >
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="content"
        >
          <div
            className="ui grid p-16 py-9 bg-white"
          >
            <div
              className="left aligned eleven wide column"
            >
              <p
                className="ui header description"
              >
                Bitcoin
              </p>
              <p
                className="ui header comment"
              >
                https://grove.chat/  -  Tue, Apr. 26, 2022 5:00 am
              </p>
            </div>
            <div
              className="middle aligned five wide column"
            >
              <div
                className="ui fitted toggle checkbox"
                onChange={[Function]}
                onClick={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
              >
                <input
                  checked={false}
                  className="hidden"
                  readOnly={true}
                  tabIndex={0}
                  type="checkbox"
                />
                <label />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="content"
        >
          <div
            className="ui grid p-16 py-9 bg-white"
          >
            <div
              className="left aligned eleven wide column"
            >
              <p
                className="ui header description"
              >
                Ethereum
              </p>
              <p
                className="ui header comment"
              >
                https://uniswap.org/  -  Mon, Apr. 25, 2022 8:00 pm
              </p>
            </div>
            <div
              className="middle aligned five wide column"
            >
              <div
                className="ui fitted toggle checkbox"
                onChange={[Function]}
                onClick={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
              >
                <input
                  checked={false}
                  className="hidden"
                  readOnly={true}
                  tabIndex={0}
                  type="checkbox"
                />
                <label />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
